### PR TITLE
Add coupled_groups to SMP

### LINKED ISSUE #5679: Add convenience parameter for groups of coupled variables to SMP
Add a parameter to ```SingleMatrixPreconditioner``` that will accept a list of couple variable groups:

```
coupled_groups = 'c,w,eta disp_x,gxx,gxy disp_y,gyx,gyy'
```

This will be a more convenient way of declaring the off-diagonal jacobian blocks for the preconditioning matrix for many phase field applications, where we have lots of coupling between groups of variables but where a ```full = true``` would be overkill.

### PR DESCRIPTION
**Pull request by dschwen**
Adds a convenience parameter to SMP to allow specifying groups of coupled variables (for generating off-diagonal Jacobian entries).

Closes #5679

### CODE REVIEW

**Comment by friedmud** [comment]
This looks good - but can I ask one more thing while you're here?  Can you add a test for this to moose_test?

File: framework/src/preconditioners/SingleMatrixPreconditioner.C [hunk 1]
```diff
@@ -43,14 +45,30 @@ SingleMatrixPreconditioner::SingleMatrixPreconditioner(const InputParameters & p
...
-    // off-diagonal entries
+    // off-diagonal entries from the off_diag_row and off_diag_column parameters
     std::vector<std::vector<unsigned int> > off_diag(n_vars);
-    for (unsigned int i = 0; i < getParam<std::vector<std::string> >("off_diag_row").size(); i++)
+    for (unsigned int i = 0; i < getParam<std::vector<NonlinearVariableName> >("off_diag_row").size(); i++)
     {
-      unsigned int row = nl.getVariable(0, getParam<std::vector<std::string> >("off_diag_row")[i]).number();
-      unsigned int column = nl.getVariable(0, getParam<std::vector<std::string> >("off_diag_column")[i]).number();
+      unsigned int row = nl.getVariable(0, getParam<std::vector<NonlinearVariableName> >("off_diag_row")[i]).number();
+      unsigned int column = nl.getVariable(0, getParam<std::vector<NonlinearVariableName> >("off_diag_column")[i]).number();
```
Comment (friedmud) [inline framework/src/preconditioners/SingleMatrixPreconditioner.C]: Thanks for fixing this up

**Comment by dschwen** [comment]
I added the test. It is not a particularly strong test though (like the other SMP tests). I'm using a NEWTON solve and limited the max nl iterations to 2. The test fails if you disable off diagonal jacobians.

**Comment by friedmud** [comment]
@dschwen thanks for adding the test.  At least if we do something horribly wrong it should break ;-)

**Pull request merged**